universal-application-tool-0.0.1/app/services/program/BlockDefinition.java

```diff
@@ -0,0 +1,54 @@
+package services.program;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import services.question.QuestionDefinition;
+
+/** Defines a single program block, which contains a list of questions and data about the block. */
+@AutoValue
+public abstract class BlockDefinition {
```

Comment:
suggestion (blocking): We'll want an ID for blocks as well (see https://github.com/seattle-uat/universal-application-tool/pull/117 for discussion)

Mind adding a `long id()` to the class?